integrationtests/gquic/drop_test.go

```diff
@@ -22,74 +23,115 @@ var directions = []quicproxy.Direction{quicproxy.DirectionIncoming, quicproxy.Di
 var _ = Describe("Drop tests", func() {
 	var proxy *quicproxy.QuicProxy
 
-	runDropTest := func(dropCallback quicproxy.DropCallback, version protocol.VersionNumber) {
+	startProxy := func(dropCallback quicproxy.DropCallback, version protocol.VersionNumber) {
 		var err error
 		proxy, err = quicproxy.NewQuicProxy("localhost:0", version, &quicproxy.Opts{
 			RemoteAddr: "localhost:" + testserver.Port(),
 			DropPacket: dropCallback,
 		})
 		Expect(err).ToNot(HaveOccurred())
+	}
 
+	downloadFile := func(version protocol.VersionNumber) {
 		command := exec.Command(
 			clientPath,
 			"--quic-version="+strconv.Itoa(int(version)),
 			"--host=127.0.0.1",
 			"--port="+strconv.Itoa(proxy.LocalPort()),
 			"https://quic.clemente.io/prdata",
 		)
-
 		session, err := Start(command, nil, GinkgoWriter)
 		Expect(err).NotTo(HaveOccurred())
 		defer session.Kill()
 		Eventually(session, 20).Should(Exit(0))
 		Expect(bytes.Contains(session.Out.Contents(), testserver.PRData)).To(BeTrue())
 	}
 
+	downloadHello := func(version protocol.VersionNumber) {
+		command := exec.Command(
+			clientPath,
+			"--quic-version="+strconv.Itoa(int(version)),
+			"--host=127.0.0.1",
+			"--port="+strconv.Itoa(proxy.LocalPort()),
+			"https://quic.clemente.io/hello",
+		)
+		session, err := Start(command, nil, GinkgoWriter)
+		Expect(err).NotTo(HaveOccurred())
+		defer session.Kill()
+		Eventually(session, 20).Should(Exit(0))
+		Expect(session.Out).To(Say(":status 200"))
+		Expect(session.Out).To(Say("body: Hello, World!\n"))
+	}
+
+	deterministicDropper := func(p, interval, dropInARow uint64) bool {
+		return (p % interval) < dropInARow
+	}
+
+	stochasticDropper := func(freq int) bool {
+		return mrand.Int63n(int64(freq)) == 0
+	}
+
 	AfterEach(func() {
 		Expect(proxy.Close()).To(Succeed())
 	})
 
-	Context("after the crypto handshake", func() {
-		deterministicDropper := func(p, interval, dropInARow uint64) bool {
-			if p <= 10 { // don't interfere with the crypto handshake
-				return false
-			}
-			return (p % interval) < dropInARow
-		}
-
-		stochasticDropper := func(p protocol.PacketNumber, freq int) bool {
-			if p <= 10 { // don't interfere with the crypto handshake
-				return false
-			}
-			return mrand.Int63n(int64(freq)) == 0
-		}
-
-		for _, v := range protocol.SupportedVersions {
-			version := v
-
-			Context(fmt.Sprintf("with QUIC version %d", version), func() {
+	for _, v := range protocol.SupportedVersions {
+		version := v
+
+		Context(fmt.Sprintf("with QUIC version %d", version), func() {
+			Context("during the crypto handshake", func() {
+				for _, d := range directions {
+					direction := d
+
+					It("establishes a connection when the first packet is lost", func() {
```

Comment:
The direction should be in the It string. Also in the next 2 cases.

Reply:
Done.